### PR TITLE
HSEARCH-3514 Add automatic Java module names and test them

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-3514

### CODE REVIEW

**Comment by yrodiere** [comment]
On the chat, you mentioned an exception when trying to generate the module-info for the ORM mapper. I don't see it here. Did you fix it? Is this PR ready for review?

**Comment by yrodiere** [comment]
I know there's still a "Preview" label, but everything looked like it worked, so I'm a bit confused.

**Comment by fax4ever** [comment]
I tried to apply all the steps described in the issue page.

I left 2 TODO that, in my opinion, they should be addressed in HSEARCH‌-3274.
But if you prefer I could handle them here, for it would be the same.

I run `SUPPORTED_ENV_ONLY` integration test special build:
http://ci.hibernate.org/job/hibernate-search/view/change-requests/job/PR-1911/6/.

File: mapper/orm/pom.xml [hunk 1]
```diff
@@ -97,4 +101,40 @@
...
     </build>
+
+    <profiles>
+        <profile>
+            <id>module-descriptors</id>
+            <!--
+                TODO HSEARCH-3274
+                Trying to generate module info, we got some issues on multi-release dependent modules.
+                Invocation of jdeps failed: jdeps -generate-module-info ...
+                Error: byte-buddy-1.9.5.jar is a multi-release jar file but -multi-release option is not set.
```
Comment (yrodiere) [inline mapper/orm/pom.xml]: I tried to investigate this too, but it seems it boils down to excluding dependencies as well.
I added a comment.

**Comment by fax4ever** [comment]
@yrodiere thank you, I think I've applied the changes here.

**Comment by yrodiere** [comment]
Merged, thanks!

**Comment by fax4ever** [comment]
@yrodiere thanks!

**Pull request merged**